shared/lib/multichain.ts

```diff
@@ -59,3 +65,15 @@ export function getCaipNamespaceFromAddress(address: string): CaipNamespace {
   // Defaults to "Ethereum" for all other cases for now.
   return KnownCaipNamespace.Eip155;
 }
+
+export function isCurrentChainCompatibleWithAccount(
+  chainId: string,
+  account: InternalAccount,
+): boolean {
+  if (isCaipChainId(chainId)) {
+    const { namespace } = parseCaipChainId(chainId);
+    return namespace === getCaipNamespaceFromAddress(account.address);
+  }
+
+  return isEvmAccountType(account.type);
```

Comment:
Maybe something slightly "safer" (just in case the chain ID is not "defined")

```suggestion
  // For EVM accounts, we do not check the chain ID format, but we just expect it
  // to be defined.
  return isEvmAccountType(account.type) && Boolean(chainId);
```